verification/verify.go

```diff
@@ -0,0 +1,133 @@
+package verification
+
+import (
+	"errors"
+	"sort"
+
+	"github.com/livepeer/go-livepeer/core"
+	"github.com/livepeer/go-livepeer/net"
+
+	"github.com/livepeer/lpms/ffmpeg"
+	"github.com/livepeer/lpms/stream"
+)
+
+// Special error type indicating a retryable error
+// Such errors typically mean re-trying the transcode might help
+// (Non-retryable errors usually indicate unrecoverable system errors)
+type Retryable struct {
+	error
+}
+
+var ErrPixelMismatch = Retryable{errors.New("PixelMismatch")}
+
+type VerifierParams struct {
+	// ManifestID should go away once we do direct push of video
+	ManifestID core.ManifestID
+
+	// Bytes of the source video segment
+	Source *stream.HLSSegment
+
+	// Rendition parameters to be checked
+	Profiles []ffmpeg.VideoProfile
+
+	// Information on the orchestrator that performed the transcoding
+	Orchestrator *net.OrchestratorInfo
+
+	// Transcoded result metadata
+	Results *net.TranscodeData
+}
+
+type VerificationResult interface {
```

Comment:
Why use an interface here? Do you imagine implementations that do more than just return internal score and pixels fields (as is done by the `verificationResult` type)?

Reply:
Yeah, an interface is not really necessary here. Removed in b70f22f26f926415ae39a887ca1d74eb7646c1cd

Reply:
The linter was complaining about "stuttering" (`verification.VerificationResult`), and suggested just `Result` so I went with that (`Results` actually), the brevity is much better. Also took the liberty of renaming `VerifierParams` to simply `Params` as well for consistency. https://github.com/livepeer/go-livepeer/pull/1096/commits/cbe81d9208d5f65c63e9914c78781156747c9cc0